src/Core/Domain/Product/Command/UpdateProductStockCommand.php

```diff
@@ -0,0 +1,392 @@
+    {
+        return $this->useAdvancedStockManagement;
+    }
+
+    /**
+     * @param bool $useAdvancedStockManagement
+     *
+     * @return UpdateProductStockCommand
+     */
+    public function setUseAdvancedStockManagement(bool $useAdvancedStockManagement): UpdateProductStockCommand
+    {
+        $this->useAdvancedStockManagement = $useAdvancedStockManagement;
+
+        return $this;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function dependsOnStock(): ?bool
+    {
+        return $this->dependsOnStock;
+    }
+
+    /**
+     * @param bool $dependsOnStock
+     *
+     * @return UpdateProductStockCommand
+     */
+    public function setDependsOnStock(bool $dependsOnStock): UpdateProductStockCommand
+    {
+        $this->dependsOnStock = $dependsOnStock;
+
+        return $this;
+    }
+
+    /**
+     * @return PackStockType|null
+     */
+    public function getPackStockType(): ?PackStockType
+    {
+        return $this->packStockType;
+    }
+
+    /**
+     * @param string $packStockType
+     *
+     * @return UpdateProductStockCommand
+     *
+     * @throws ProductPackConstraintException
+     */
+    public function setPackStockType(string $packStockType): UpdateProductStockCommand
+    {
+        $this->packStockType = new PackStockType($packStockType);
+
+        return $this;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getQuantity(): ?int
+    {
+        return $this->quantity;
+    }
+
+    /**
+     * @param int $quantity
+     *
+     * @return UpdateProductStockCommand
+     */
+    public function setQuantity(int $quantity): UpdateProductStockCommand
+    {
+        $this->quantity = $quantity;
+
+        return $this;
+    }
+
+    /**
+     * @return OutOfStockType|null
+     */
+    public function getOutOfStockType(): ?OutOfStockType
+    {
+        return $this->outOfStockType;
+    }
+
+    /**
+     * @param string $outOfStockType
+     *
+     * @return $this
+     *
+     * @throws ProductStockConstraintException
+     */
+    public function setOutOfStockType(string $outOfStockType): UpdateProductStockCommand
+    {
+        $this->outOfStockType = new OutOfStockType($outOfStockType);
+
+        return $this;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getMinimalQuantity(): ?int
+    {
+        return $this->minimalQuantity;
+    }
+
+    /**
+     * @param int $minimalQuantity
+     *
+     * @return UpdateProductStockCommand
+     */
+    public function setMinimalQuantity(int $minimalQuantity): UpdateProductStockCommand
+    {
+        $this->minimalQuantity = $minimalQuantity;
+
+        return $this;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getLocation(): ?string
+    {
+        return $this->location;
+    }
+
+    /**
+     * @param string $location
+     *
+     * @return UpdateProductStockCommand
+     */
+    public function setLocation(string $location): UpdateProductStockCommand
```

Comment:
Should we rename Command to UpdateProductStockSettingsCommand ?

When I see UpdateProductStockCommand, I think "oh, this is used to update the stock quantity !" 😄

Reply:
It does update the quantity Among other things

Reply:
I understand why you suggest a more detailed name since it's more than updating one value However all other commands reflect this idea:
- `UpdateProductSeoCommand` (not `UpdateProductSeoSettingsCommand`)
- `UpdateProductShippingCommand` (not `UpdateProductShippingInformationsCommand`)

And so on, they each relate to a subdomain so once you get the big picture it actually makes sense and it allows to keep short understandable names.

I almost decided to follow your lead because I though maybe we will need a command to change the stock quantity only and the name would already be taken! But then I figured then we would call such command `UpdateProductQuantityCommand` because Stock is actually a wider domain than just quantity So I think the naming suits the need in this case 🤷‍♂️

Reply:
> I almost decided to follow your lead

So close! 💪 

But OK for keeping current name 👍